### PR TITLE
Forward the errors from the request on the prerender server.

### PR DESCRIPTION
**Pull request by alexandrusavin**
Make sure that the afterRenderFn is called even if Request returns an error.

### CODE REVIEW

**Comment by alexandrusavin** [comment]
It's related to (would give that posibility)

**Comment by alexandrusavin** [comment]
This is how one would catch errors if the prerender server fails:

```javascript
app.use(require('prerender-node').set('afterRender', function(req, prerender_res) {
  if (prerender_res instanceof Error) {
    // the prerender server is not available or some error occurred in the proxy request
  }
}))
```
Although I would prefer having the Error object in the callback in a separate variable, like this: ```function(err, req, prerender_res)```

**Comment by thoop** [comment]
Thanks! I would say it would probably be better to have the error in the callback as a separate variable. Since the error is tied to the response, would it make more sense as `function(req, err, prerender_res)`? I don't know if I've ever seen a callback with the error as the second param though.

**Comment by alexandrusavin** [comment]
Yeah ... I was thinking also on which is the best way to place it. The error is part of the result context and should stay before the prerender_res. On the other had, the convention used by the node core is that the first place in ANY callback is reserved for the error object.

And wouldn't any of these changes be a breaking one for other users of the module?

**Comment by alexandrusavin** [comment]
I think the error object should be in the first position because other generic modules might expect that as a standard. Failing to do this would brake the generic part of the callback and would imply custom code to handle this (as it would now).

But adding it to the first position would also definitely be a braking change for the other users of this module.

File: index.js [hunk 1]
```diff
@@ -135,15 +136,15 @@ prerender.shouldShowPrerenderedPage = function(req) {
 
   //if it is a bot and not requesting a resource and is not blacklisted(url or referer)...dont prerender
   if(Array.isArray(this.blacklist) && this.blacklist.some(function(blacklisted){
-    var blacklistedUrl = false
-      , blacklistedReferer = false
-      , regex = new RegExp(blacklisted);
+      var blacklistedUrl = false
```
Comment (homeyer) [inline index.js]: unnecessary whitespace changes, let's clean this up.

**Comment by homeyer** [comment]
I'd vote to have err be the first arg as well. This PR is also missing tests.
